src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1104,6 +1103,155 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
         }
     }
 
+    @Tidy
+    def "System doesn't reroute flow to more preferable path while updating portNumber/vlanId"() {
+        given: "Two active switches connected to traffgens with two possible paths at least"
+        def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected*.dpId ?:
+                assumeTrue("Should be at least two active traffgens connected to switches", false)
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { swP ->
+            allTraffgenSwitchIds.contains(swP.src.dpId) && allTraffgenSwitchIds.contains(swP.dst.dpId) &&
+                    swP.paths.size() >= 2
+        } ?: assumeTrue("Unable to find required switches/paths in topology",false)
+
+        and: "A flow"
+        def flow = flowHelperV2.randomFlow(switchPair, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "Make the current path less preferable than alternatives"
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+        def alternativePaths = switchPair.paths.findAll { it != currentPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+
+        and: "Update the flow: port number and vlanId on the src/dst endpoints"
+        def updatedFlow = flow.jacksonCopy().tap {
+            it.source.portNumber = 10 // traffgen port
+            it.source.vlanId = flow.destination.vlanId + 1
+            it.destination.portNumber = 10 // traffgen port
+            it.destination.vlanId = flow.source.vlanId + 1
+        }
+        flowHelperV2.updateFlow(flow.flowId, updatedFlow)
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == updatedFlow.source.portNumber
+            it.source.vlanId == updatedFlow.source.vlanId
```

Comment:
destination is updated on line 1129, no checks for it is made here

Reply:
oops, my bad.
fixed